src/client/dfuse/dfuse_core.c

```diff
@@ -432,6 +468,170 @@ dfuse_pool_open(struct dfuse_projection_info *fs_handle, uuid_t *pool,
 	return rc;
 }
 
+#define ATTR_COUNT 6
+
+char const *const
+cont_attr_names[ATTR_COUNT] = {"dfuse-attr-time",
+			       "dfuse-dentry-time",
+			       "dfuse-dentry-dir-time",
+			       "dfuse-ndentry-time",
+			       "dfuse-data-cache",
+			       "dfuse-direct-io-disable"};
+
+#define ATTR_TIME_INDEX		0
+#define ATTR_DENTRY_INDEX	1
+#define ATTR_DENTRY_DIR_INDEX	2
+#define ATTR_NDENTRY_INDEX	3
+#define ATTR_DATA_CACHE_INDEX	4
+#define ATTR_DIRECT_IO_DISABLE_INDEX	5
+
+/* Attribute values are of the form "120M", so the buffer does not need to be
+ * large.
+ */
+#define ATTR_VALUE_LEN 128
+
+/* Setup caching attributes for a container.
+ *
+ * These are read from pool attributes, or can be overwritten on the command
+ * line, but only for the root dfc in that case, so to use caching with
+ * multiple containers it needs to be set via attributes.
+ *
+ * Returns a  error code on error, or ENODATA if no attributes are
+ * set.
+ */
+static int
+dfuse_cont_get_cache(struct dfuse_cont *dfc)
+{
+	size_t		size;
+	char		*buff;
+	int		rc;
+	int		i;
+	unsigned int	value;
+	bool		have_dentry = false;
+	bool		have_dentry_dir = false;
+	bool		have_attr = false;
+	bool		have_dio = false;
+	bool		have_cache_off = false;
+
+	D_ALLOC(buff, ATTR_VALUE_LEN);
+	if (buff == NULL)
+		return ENOMEM;
+
+	for (i = 0; i < ATTR_COUNT; i++) {
+		size = ATTR_VALUE_LEN - 1;
+
+		rc = daos_cont_get_attr(dfc->dfs_coh, 1, &cont_attr_names[i],
```

Comment:
we could get all attributes in a single call, no?

Reply:
I don't think we can yet, but @bfaccini is working on it.  At the minute we could only get all attributes in a single call if they all exist. https://jira.hpdd.intel.com/browse/DAOS-6710

We'll update this code once the feature is available.